nemo/core/config/pytorch_lightning.py

```diff
@@ -13,39 +13,82 @@
 # limitations under the License.
 
 from dataclasses import dataclass
-from typing import Any, Dict, List, Optional
+from typing import Any, Dict, List, Optional, Union
+
+from hydra.core.config_store import ConfigStore
+from hydra.types import ObjectConf
 
 __all__ = ['TrainerConfig']
 
 
+cs = ConfigStore.instance()
+
+
 @dataclass
 class TrainerConfig:
```

Comment:
This is going to be really hard to keep up to date with PTL as they evolve. Is it possible to ask them for a commitment to update this on their point releases?

Reply:
This is a longer topic. In short: in the future this will be part of PTL. But until PTL-hydra PR is blocked, need to keep it here.

Note in this PR I have updated list of trainer params to the latest PTL version.

Reply:
@blisc, we can do a check or a unit test somewhere to ensure this configs keys set matches exactly the key spec of PTL trainer (using inspect). Raise an error otherwise